tests/test_validate.py

```diff
@@ -350,3 +352,39 @@ def test_validate_dictionnary_no_multiprocess_compatible_warning():
     assert data['metadata']['version'] == u'1.0.0'
     assert data['metadata']['listed'] is True
     assert data['metadata']['id'] == u'my@dict'
+
+def test_validate_hybrid_addon_xpi():
+    """Integration test for the expected notice message on hybrid add-ons."""
+    result = validate(path='tests/resources/validate/hybrid_extension.xpi')
+    data = json.loads(result)
+
+    assert data['success'] is True
+    assert data['errors'] == 0
+    assert data['notices'] == 1
+    assert data['warnings'] == 0
+    assert data['compatibility_summary']
+    assert data['compatibility_summary']['errors'] == 0
+    assert data['compatibility_summary']['notices'] == 0
+    assert data['compatibility_summary']['warnings'] == 0
+    assert data['detected_type'] == 'extension'
+
+    assert len(data['messages']) == 1
+    assert data['messages'][0]['file'] == u'install.rdf'
+    assert data['messages'][0]['message'] == u'This add-on contains an embedded webextension'
+
+    expectedHybridDescriptionRegExp = re.compile (
+        '^The embedded webextension is in the folder "webextension"\. .*'
+        'For more information, refer to https://developer.mozilla.org/.*Embedded_WebExtensions$'
+    )
+    assert expectedHybridDescriptionRegExp.match(data['messages'][0]['description'])
```

Comment:
This could be simplified to just do something like `assert 'The embedded webextension is in the folder "webextension"' in data['messages'][0]['description']`